tests/tools/go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/signalfx/splunk-otel-collector/tests/tools
 
-go 1.20
+go 1.21
+
+toolchain go1.21.1
 
```

Comment:
This should probably be done in another PR (unless I'm missing something)

Reply:
It was auto generated for my machine when I did go tidy I believe.

Reply:
I think if you undo the mod changes, `export GOTOOLCHAIN=local`, and make tidy it won't restrict for others?

we probably need to add this as a default env var for make.